### PR TITLE
Add Foreman Git templates plugin

### PR DESCRIPTION
**Pull request by dgoetz**
Depends on https://github.com/theforeman/puppet-foreman/pull/1059

### CODE REVIEW

**Comment by upadhyeammit** [comment]
@dgoetz rebase please

hmm i think it will also require release of new puppet-foreman

**Comment by ekohl** [comment]
[test foreman-installer]

The installer should use git builds for our modules. However, when this PR was submitted the dependency wasn't merged yet. I expect it to pass now.

File: config/foreman.migrations/20220602164000_foreman_git_templates.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+answers['foreman::plugin::git_templates'] ||= false
```
Comment (ekohl) [inline config/foreman.migrations/20220602164000_foreman_git_templates.yaml]: The migration should be a `.rb` file, not `.yaml`. Same with the other file.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: config/foreman.migrations/20220602164000_foreman_git_templates.yaml (hunk 1 above)
Comment (dgoetz) [reply config/foreman.migrations/20220602164000_foreman_git_templates.yaml]: Renamed it here and in all other pull requests (except the first one where I did it correct).

**Review by ekohl — APPROVED** [review]
Thanks!

**Comment by ekohl** [comment]
Looks like you'll need to adjust the tests too.

**Comment by dgoetz** [comment]
I adjusted the tests (as the migration now really works, tests had to be adjusted). This one is already green, others PRs tests are running and should become green. If some rebasing is needed just drop me a comment. Thanks for reviewing!

**Review by ekohl — APPROVED** [review]

**Pull request merged**